README.md

```diff
@@ -55,10 +55,120 @@ console.log(hi); // => howdy world!
+
+Note: for namespaced modules the path will end up being `s3://mapbox-node-binary/@org/<your module name>`.
+
+2) Copy the ci.template.js
+
+Copy the `ci.template.js` from this repo into your repo and place it at  `./cloudformation/ci.template.js`
+
+3) Install deps
+
+```
+npm install @mapbox/cfn-config
+npm install @mapbox/cloudfriend
+```
+
+4) Create a user with permissions to upload to `s3://mapbox-node-binary/<your module name>/`
+
+First configure your AWS creds. You will need to set at least the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` keys. And also `AWS_SESSION_TOKEN` if you are using 2-factor auth.
+
+Then run:
+
+```bash
+./node_modules/.bin/validate-template cloudformation/ci.template.js
+./node_modules/.bin/build-template cloudformation/ci.template.js > cloudformation/ci.template
+```
+
+Next we will actually create the user. But first let's discuss what happens here. In addition to creating the user we also write details about the user to a separate bucket (for east auditing purposes). In the below command we:
+
+  - Create a `ci-binary-publish` user
+  - Using the `cloudformation/ci.template`
+  - And ask to save our stack configuration to the `cfn-config` bucket. If you are outside of Mapbox, see [you will need to pass a different bucket and also a `--template-bucket` option](https://github.com/mapbox/cfn-config#prerequisites)
+
+Now, run the command to create the user:
+
+```
+./node_modules/.bin/cfn-config create ci-binary-publish cloudformation/ci.template -c cfn-configs
+```
+
+It will prompt you, choose:
+
+  - New configuration
+  - Ready to create the stack? Y
+
+It will fail if the stack already exists. In this case you can recreate a new user by deleting the stack by running `./node_modules/.bin/cfn-config delete ci-binary-publish cloudformation/ci.template` and then creating a new one.
+
+5) Get the user keys
+
+After the create step succeeds you will have a new user. You now need to get get the users `AccessKeyId` and `SecretAccessKey`.
+
+You can do this by:
+
+ - Go to https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks?filter=active&tab=outputs
+ - Search for `ci-binary-publish`. You should see a stack called `<your module name>-ci-binary-publish`
+ - Click the checkbox beside your `<your module name>-ci-binary-publish` stack
+ - Click the `Output` tab to access the `AccessKeyId` and `SecretAccessKey` for this new user.
+
+6) Add the keys to the travis
+
+You can do this two ways: 1) add the keys to the travis UI settings, or 2) encode them as secure variables in your `.travis.yml`
+
+#### Adding to travis UI settings
+
+- Go to https://travis-ci.org/<your user or org>/<your module>/settings
+- Scroll to the bottom and find the `Environment Variables` section
+- Add a variable called `AWS_ACCESS_KEY_ID` and put the value of the `AccessKeyId` in it
+- CRITICAL: Choose `OFF` for `Display value in build log` to ensure the variables are now shown in the logs
```

Comment:
We don't want to show our access keys in the logs on travis, right? I think we should keep critical to `ON` here. The logs will still show which variables exist, but if critical is turned `OFF` it will show the variable AND the value of that variable.